testing/file_test/test_file.cpp

```diff
@@ -93,9 +96,47 @@ struct SplitState {
   int file_index = 0;
 };
 
+static auto ExtractFilePathFromUri(llvm::StringRef uri) -> llvm::StringRef {
+  static constexpr llvm::StringRef FilePrefix = "file:/";
+  if (uri.starts_with(FilePrefix)) {
+    return uri.drop_front(FilePrefix.size());
+  }
+  return uri;
+}
+
+static auto AutoFillDidOpenParams(llvm::json::Object* params,
+                                  llvm::ArrayRef<TestFile::Split> splits)
+    -> ErrorOr<Success> {
+  auto* text_document = params->getObject("textDocument");
+  if (text_document == nullptr) {
+    return Success{};
+  }
+
+  auto attr_it = text_document->find("text");
+  if (attr_it == text_document->end() || attr_it->second != "AUTOFILL") {
+    return Success{};
+  }
+
+  auto uri = text_document->getString("uri");
+  CARBON_CHECK(uri.has_value());
+
+  auto file_path = ExtractFilePathFromUri(*uri);
```

Comment:
Is the url allowed to be just a file without the prefix? Right now that would be accepted. Maybe ExtractFilePathFromUri should return an optional?

Reply:
True a uri without file scheme should not be accepted, I will restrict this further to return an error.